### DISCUSSION TITLE
How to get all existing keys on redis?

### DISCUSSION THREAD

**Discussion by berkayalan** [redis/go-redis#3147] [Q&A]
Hey, I want to get all existing keys but commands like "keys * " does not work. Any idea how I can do that on go? Thanks

**Comment by themartorana** [comment]
Use `SCAN` and you should be able to iterate over all keys. Here's a guide: https://redis.uptrace.dev/guide/get-all-keys.html

**Comment by berkayalan** [comment]
Thanks, this logic works.
